scss/_buttons.scss

```diff
@@ -43,38 +57,82 @@
   &.disabled,
   fieldset:disabled & { // stylelint-disable-line selector-no-qualifying-type
     pointer-events: none;
+    // Remove CSS gradients if they're enabled
+    background-image: if($enable-gradients, none, null);
     opacity: $btn-disabled-opacity;
     @include box-shadow(none);
   }
 }
 
+.btn-outline {
+  color: var(--btn-accent-color);
+  background-color: transparent;
+  border-color: currentColor;
+
+  &:hover {
+    background-color: var(--btn-accent-color);
+    border-color: var(--btn-accent-color);
+  }
+
+  &:focus,
+  &.focus {
+    color: var(--btn-accent-contrast-color);
+    background-color: var(--btn-accent-color);
+    border-color: var(--btn-accent-color);
+  }
+
+  &.active,
+  &:active {
+    color: var(--btn-accent-contrast-color);
+    background-color: var(--btn-accent-color);
+    border-color: var(--btn-accent-color);
+  }
+}
```

Comment:
If I am not mistaken, the approach would be something like,

```css
.btn {
  --btn-color: default here;
  --btn-background-color: default here;
  --btn-border-color: currentColor;

  color: var(--btn-color);
  background-color: var(--btn-background-color);
  border-color: var(--btn-border-color);
}

.btn-outline {
  --btn-color: change me;
  --btn-background-color: change me;
  --btn-border-color: change me;

  &:hover {
     --btn-color: change me;
     --btn-background-color: change me;
     --btn-border-color: change me;
  }

  &:focus,
  &.focus {
      --btn-color: default here;
      --btn-background-color: default here;
      --btn-border-color: currentColor;
  }

  &.active,
  &:active {
      --btn-color: change me;
      --btn-background-color: change me;
      --btn-border-color: change me;
  }
}
```

the idea is that you focus more on swapping the values than overriding styles.

At least that is how I understand it.